SDK/MobileCenterRum/Microsoft.Azure.Mobile.Rum.Windows.Shared/RealUserMeasurements.cs

```diff
@@ -199,180 +203,209 @@ private void DisposeCancellationSource()
             _cancellationTokenSource = null;
         }
 
-        private async Task RunTestsAsync(string rumKey, CancellationToken cancellationToken)
+        private async Task RunMeasurementsAsync(string rumKey, CancellationToken cancellationToken)
         {
             // TODO handle network state, requires refactoring to reuse ingestion logic here
             using (var httpNetworkAdapter = new HttpNetworkAdapter())
             {
                 // Get remote configuration.
-                var jsonConfiguration = await httpNetworkAdapter.SendAsync($"{ConfigurationEndpoint}/{ConfigurationFileName}", HttpMethod.Get, Headers, "", cancellationToken);
+                var jsonConfiguration = await GetRemoteConfigurationAsync(cancellationToken, httpNetworkAdapter);
 
                 // Parse configuration.
                 var configuration = JsonConvert.DeserializeObject<RumConfiguration>(jsonConfiguration);
 
-                // Prepare random weighted selection.
-                var configurationOk = false;
+                // If configuration looks good so far.
                 if (configuration.TestEndpoints != null)
                 {
-                    var totalWeight = 0;
-                    var weightedEndpoints = new List<TestEndpoint>(configuration.TestEndpoints.Count);
-                    foreach (var endpoint in configuration.TestEndpoints)
+                    // Generate test urls.
+                    var testUrls = GenerateTestUrls(configuration);
+
+                    // Run the tests.
+                    await RunTestUrlsAsync(cancellationToken, httpNetworkAdapter, testUrls);
+
+                    // Report the results.
+                    await ReportMeasurementsAsync(rumKey, cancellationToken, httpNetworkAdapter, configuration, testUrls);
+                }
+                else
+                {
+                    throw new MobileCenterException("Configuration does not contain test endpoints.");
+                }
+            }
+        }
+
+        private async Task<string> GetRemoteConfigurationAsync(CancellationToken cancellationToken, IHttpNetworkAdapter httpNetworkAdapter)
+        {
+            foreach (var url in ConfigurationEndpoints)
+            {
+                try
+                {
+                    MobileCenterLog.Verbose(LogTag, "Calling " + url);
+                    return await httpNetworkAdapter.SendAsync(url, HttpMethod.Get, Headers, "", cancellationToken);
+                }
+                catch (OperationCanceledException)
```

Comment:
Might be a good case to use C# 6 exception filters?

`try {...} catch (Exception e) when (!(e is OperationCanceledException)) {...}`

Reply:
Fixed